jetty-ee10/jetty-ee10-servlet/src/test/java/org/eclipse/jetty/ee10/servlet/DefaultServletCombinationsTest.java

```diff
@@ -0,0 +1,258 @@
+
+    public Path docRoot;
+    private Server server;
+    private LocalConnector connector;
+    private ServletContextHandler context;
+
+    @BeforeEach
+    public void init() throws Exception
+    {
+        docRoot = workDir.getEmptyPathDir().resolve("docroot");
+        FS.ensureDirExists(docRoot);
+        Files.writeString(docRoot.resolve("index.html"), "Static index.html at root", UTF_8);
+        Files.writeString(docRoot.resolve("foo.welcome"), "Static foo.welcome at root", UTF_8);
+
+        Path subdirPath = docRoot.resolve("subdir");
+        FS.ensureDirExists(subdirPath);
+        Files.writeString(subdirPath.resolve("index.html"), "Static index.html at root subdir", UTF_8);
+        Files.writeString(subdirPath.resolve("foo.welcome"), "Static foo.welcome at root subdir", UTF_8);
+
+        Path emptyPath = docRoot.resolve("empty");
+        FS.ensureDirExists(emptyPath);
+
+        Path staticPath = docRoot.resolve("static");
+        FS.ensureDirExists(staticPath);
+        Files.writeString(staticPath.resolve("index.html"), "Static index.html at static", UTF_8);
+        Files.writeString(staticPath.resolve("foo.welcome"), "Static foo.welcome at static", UTF_8);
+
+        Path staticsubdirPath = staticPath.resolve("subdir");
+        FS.ensureDirExists(staticsubdirPath);
+        Files.writeString(staticsubdirPath.resolve("index.html"), "Static index.html at static subdir", UTF_8);
+        Files.writeString(staticsubdirPath.resolve("foo.welcome"), "Static foo.welcome at static subdir", UTF_8);
+
+        Path subdirHtmlPath = docRoot.resolve("subdirHtml");
+        FS.ensureDirExists(subdirHtmlPath);
+        Files.writeString(subdirHtmlPath.resolve("index.html"), "Static index.html at root subdirHtml", UTF_8);
+        Path staticSubdirHtmlPath = staticPath.resolve("subdirHtml");
+        FS.ensureDirExists(staticSubdirHtmlPath);
+        Files.writeString(staticSubdirHtmlPath.resolve("index.html"), "Static index.html at static subdirHtml", UTF_8);
+
+        Path subdirWelcomePath = docRoot.resolve("subdirWelcome");
+        FS.ensureDirExists(subdirWelcomePath);
+        Files.writeString(subdirWelcomePath.resolve("foo.welcome"), "Static foo.welcome at root subdirWelcome", UTF_8);
+
+        Path staticSubdirWelcomePath = staticPath.resolve("subdirWelcome");
+        FS.ensureDirExists(staticSubdirWelcomePath);
+        Files.writeString(staticSubdirWelcomePath.resolve("foo.welcome"), "Static foo.welcome at static subdirWelcome", UTF_8);
+
+        Path subdirEmptyPath = staticPath.resolve("subdirEmpty");
+        FS.ensureDirExists(subdirEmptyPath);
+
+        server = new Server();
+
+        connector = new LocalConnector(server);
+        connector.getConnectionFactory(HttpConfiguration.ConnectionFactory.class).getHttpConfiguration().setSendServerVersion(false);
+
+        context = new ServletContextHandler();
+        context.setBaseResourceAsPath(docRoot);
+        context.setContextPath("/");
```

Comment:
let's have a real context to check we are not leaving it out anywhere:
```suggestion
        context.setContextPath("/ctx");
```

Reply:
Done.